proposals/semi-auto-properties.md

```diff
@@ -10,6 +10,228 @@ Two common scenarios are that you want to apply a constraint on the setter, ensu
 
 In these cases by now you always have to create an instance field and write the whole property yourself.  This not only adds a fair amount of code, but it also leaks the `field` into the rest of the type's scope, when it is often desirable to only have it be available to the bodies of the accessors.
 
+## Detailed design
+
+For properties with an `init` accessor, everything that applies below to `set` would apply instead to the `init` accessor.
+
+**Principle 1:** Every property can be thought of as having a backing field by default, which is elided when not used. The field is referenced using the keyword `field` and its visibility is scoped to the accessor bodies.
+
+**Principle 2:** `get;` will now be considered syntactic sugar for `get => field;`, and `set;` will now be considered syntactic sugar for `set => field = value;`.
+
+This means that properties may now mix and match auto accessors with full accessors. For example:
+
+```cs
+{ get; set => Set(ref field, value); }
+```
+
+```cs
+{ get => field ?? GetDefault(); set; }
```

Comment:
Nit: 

```suggestion
{ get => field ??= GetDefault(); set; }
```

Wouldn't the `??=` variant make more sense here?

Reply:
Yes, it's a typo! Thanks! I'll also take the opportunity to say `GetAmbientDefault()` to clarify the use case.

Reply:
Deleted my previous reply, I wasn't thinking. I was thinking of this as more like `GetCurrentAmbientValue()`, e.g. `?? parent?.Foo`. I'll clarify.